gatsby/src/pages/search.tsx

```diff
@@ -1,14 +1,309 @@
-import React from "react";
+import React, { useState, useEffect } from "react";
+import { Link } from "gatsby";
+import { useLocation } from "@reach/router";
+import { Card } from "@nice-digital/nds-card";
+import { SimplePagination } from "@nice-digital/nds-simple-pagination";
+import { PageHeader } from "@nice-digital/nds-page-header";
+
 import { Layout } from "../components/Layout/Layout";
 import { SEO } from "../components/SEO/SEO";
+import { setTimeout } from "timers";
+
+interface SearchResults {
+	failed: boolean;
+	resultCount: number;
+	firstResult: number;
+	lastResult: number;
+	finalSearchText: string;
+	finalSearchTextNoStopWords: string;
+	pageSize: number;
+	originalSearch: {
+		searchText: string;
+	};
+	pagerLinks: {
+		previous: {
+			fullUrl: string;
+		} | null;
+		next: {
+			fullUrl: string;
+		} | null;
+		pages: {
+			title: string;
+			url: {
+				fullUrl: string;
+			};
+			isCurrent: boolean;
+		}[];
+	};
+	documents: Document[];
+}
+
+type Document = {
+	id: string;
+	metaDescription: string;
+	contentId: string;
+	pathAndQuery: string;
+	sourceUrl: string;
+	teaser: string;
+	title: string;
+	url: string;
+};
+
+function titleString(searchText: string, pageIndex: number): string {
+	return [pageIndex > 1 && `Page ${pageIndex}`, searchText, "Search results"]
+		.filter(Boolean)
+		.join(" | ");
+}
 
 const SearchPage: React.FC = () => {
+	const [data, setData] = useState<SearchResults | null>(null);
+	const [error, setError] = useState<boolean>(false);
+	const [a11yMessage, setA11yMessage] = useState<string>("");
+
+	function announce(message: string): void {
+		setA11yMessage(message);
+	}
+
+	const location = useLocation();
+
+	useEffect(() => {
+		setData(null);
+		fetch("/api/search" + location.search)
+			.then(data => data.json())
+			.then(results => {
+				setError(false);
+				setData(results as SearchResults);
+				announce("Search results loaded");
+			})
+			.catch(() => {
+				setError(true);
+				announce("There was an error getting search results");
+			});
+	}, [location]);
+
 	return (
 		<Layout>
-			<SEO title="Search" />
-			<h1>Search</h1>
+			<div className="visually-hidden" aria-live="polite">
+				{a11yMessage}
+			</div>
+			{!error && !data && (
+				<PageHeader heading="Search results" lead="Loading" />
+			)}
+			{error && (
+				<PageHeader
+					heading="Error"
+					lead="We are currently experiencing issues with search. The issue will be
+resolved as soon as possible. We apologise for any inconvenience caused.
+"
+				/>
+			)}
+			{data && data.failed && <div>Failed to load results</div>}
```

Comment:
Should `data.failed` being true essentially be the same as error? One is we've been able to connect to the server but the server can't get results from elastic. The other is can't connect to the server. To a user I think it's the same thing.

Reply:
Ah cool had sorted this yesterday and not committed until this morn. Axe caught this in TC which was handy actually.